### PR TITLE
961 optimise reading of unprocessed AGS

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Change code to reading unprocessed AGS directly from the DB .... which didn't actually help much
- Use offset & limit to reduce the number returned ... which also didn't actually help much
- Change to BasicAssetGroupSightingSchema rather than the AsSighting schema with new fields added
    - It was this one that took it down from a couple of minutes to just over a second

### CODE REVIEW

File: app/modules/asset_groups/schemas.py [hunk 1]
```diff
@@ -180,9 +191,6 @@ class AssetGroupSightingAsSightingSchema(ModelSchema):
     timeSpecificity = base_fields.Function(
         AssetGroupSighting.config_field_getter('timeSpecificity')
     )
-    locationId = base_fields.Function(
```
Comment (hwindsor) [inline app/modules/asset_groups/schemas.py:183]: This is not the removal of a field from the API, this is the removal of duplication, see line 204 below

File: app/modules/users/resources.py [hunk 2]
```diff
@@ -538,9 +538,12 @@ class UserAssetGroupSightings(Resource):
             },
         )
         @api.parameters(PaginationParameters())
-        @api.response(AssetGroupSightingAsSightingSchema(many=True))
+        @api.response(BaseAssetGroupSightingSchema(many=True))
+        # @api.paginate()
```
Comment (hwindsor) [inline app/modules/users/resources.py:542]: This is what @bluemellophone suggested and I couldn't get to work

File: integration_tests/test_sage_detection.py [hunk 3]
```diff
@@ -64,12 +64,16 @@ def test_create_asset_group_detection(session, codex_url, test_root, login):
...
+        'locationId': 'Tiddleywink',
+        'time': '2000-01-01T01:01:01+00:00',
+        'timeSpecificity': 'time',
         'jobs': [
             {
                 'job_id': first_job['job_id'],
                 'model': 'african_terrestrial',
                 'active': False,
                 'asset_ids': [asset_guid],
+                'start': first_job['start'],
```
Comment (hwindsor) [inline integration_tests/test_sage_detection.py:76]: why this change means that the job is started by the time this message is received is a bit of a mystery.

**Review by drewblount — COMMENTED** [review]
LGTM, maybe we should check with @brmscheiner that this reduced schema is usable for the frontend though. The main time sink fixed here was processing the AssetGroupSightingAsSighting schema, which was explicitly asked-for by frontend, so let's be sure this works. 

Ben this is just modifying the user/ unprocessed AGS endpoint, now with a reduced schema that includes time, timeSpecificity, locationID, guid, stage, and indexed/ES.

**Comment by hwindsor** [comment]
> LGTM, maybe we should check with @brmscheiner that this reduced schema is usable for the frontend though. The main time sink fixed here was processing the AssetGroupSightingAsSighting schema, which was explicitly asked-for by frontend, so let's be sure this works.
> 
> Ben this is just modifying the user/ unprocessed AGS endpoint, now with a reduced schema that includes time, timeSpecificity, locationID, guid, stage, and indexed/ES.

I had a chat with Ben at ungodly o'clock for him and he confirmed it was OK. 

I also tested it against the FE and all works.

**Review by brmscheiner — APPROVED** [review]

**Review by drewblount — APPROVED** [review]

File: app/modules/users/models.py [hunk 4]
```diff
@@ -696,25 +696,22 @@ def unprocessed_asset_groups(self):
...
-            .limit(limit)
-            .all()
+        query = (
+            db.session.query(AssetGroupSighting)
+            .join(AssetGroup)
+            .filter(AssetGroup.owner_guid == self.guid)
+            .filter(AssetGroupSighting.stage != AssetGroupSightingStage.processed)
         )
 
+        return query
```
Comment (hwindsor) [inline app/modules/users/models.py:713]: How did I miss that.... ????

File: tests/modules/users/resources/test_getting_users_info.py [hunk 5]
```diff
@@ -254,7 +254,3 @@ def test_getting_asset_group_sightings_for_user(
     assert 'time' in ags
     assert ags['time']
     assert ags['timeSpecificity'] == 'time'
-    # check time in encounters as well
```
Comment (bluemellophone) [inline tests/modules/users/resources/test_getting_users_info.py:244]: @hwindsor I wanted to check that the checks for encounters should be removed

**Review by bluemellophone — APPROVED** [review]
LGTM
